src/Tokenizer.js

```diff
@@ -223,6 +223,15 @@ export class Tokenizer {
           endEarly = true;
         }
 
+        const fencesBeginRegExp = new RegExp(`^( {0,${indent}})(\`\`\`|~~~)`);
+        if (fencesBeginRegExp.test(src)) { // Items begin with at most one code block
+          const fenceBeginCap = fencesBeginRegExp.exec(src);
+          // if End list item if found non-indented fenced code block
+          if (fenceBeginCap[1].length < indent) {
+            endEarly = true;
+          }
+        }
+
         if (!endEarly) {
           const nextBulletRegex = new RegExp(`^ {0,${Math.min(3, indent - 1)}}(?:[*+-]|\\d{1,9}[.)])((?: [^\\n]*)?(?:\\n|$))`);
           const hrRegex = new RegExp(`^ {0,${Math.min(3, indent - 1)}}((?:- *){3,}|(?:_ *){3,}|(?:\\* *){3,})(?:\\n+|$)`);
```

Comment:
```suggestion
        if (!endEarly) {
          const nextBulletRegex = new RegExp(`^ {0,${Math.min(3, indent - 1)}}(?:[*+-]|\\d{1,9}[.)])((?: [^\\n]*)?(?:\\n|$))`);
          const hrRegex = new RegExp(`^ {0,${Math.min(3, indent - 1)}}((?:- *){3,}|(?:_ *){3,}|(?:\\* *){3,})(?:\\n+|$)`);
          const fencesBeginRegex = new RegExp(`^( {0,${Math.min(3, indent - 1)}})(\`\`\`|~~~)`);
```

Then below you can do a simpler test:

```
            // End list item if found code fences
            if (fencesBeginRegex.test(line)) {
              break;
            }
```